plugins/outputs/kinesis/kinesis.go

```diff
@@ -154,26 +155,33 @@ func (k *KinesisOutput) SetSerializer(serializer serializers.Serializer) {
 	k.serializer = serializer
 }
 
-func writekinesis(k *KinesisOutput, r []*kinesis.PutRecordsRequestEntry) time.Duration {
+func (k *KinesisOutput) mockKinesisService(svc kinesisiface.KinesisAPI) {
+	k.svc = svc
+}
+
+func (k *KinesisOutput) writeKinesis(r []*kinesis.PutRecordsRequestEntry) time.Duration {
+
 	start := time.Now()
 	payload := &kinesis.PutRecordsInput{
 		Records:    r,
 		StreamName: aws.String(k.StreamName),
 	}
 
+	resp, err := k.svc.PutRecords(payload)
+	if err != nil {
+		log.Printf("E! kinesis: Unable to write to Kinesis : %s", err.Error())
+		return time.Since(start)
```

Comment:
If the service errored, then `resp` will not be set

Reply:
are you saying resp needs a nil check in addition to an err check?

Reply:
No, I'm just giving insight into how the AWS APIs tend to work.